backend/app/services/integration_service.py

```
@@ -170,6 +174,11 @@ async def update_integration(
     async def delete_integration(self, user_id: UUID, integration_id: UUID) -> bool:
         """Delete an integration."""
         try:
+            # Check if integration exists before deleting
+            existing = await self.get_integration(user_id, integration_id)
+            if not existing:
+                raise ValueError("Integration not found")
```

Ruff (0.14.1)

Abstract `raise` to an inner function

(TRY301)